Combine supported and recommended requirements (#1365)
ezsystems/developer-documentation: docs/getting_started/requirements.md

    For running on [Ibexa Cloud](https://www.ibexa.co/products/ibexa-cloud), where recommended configuration and support is provided out of the box, see separate [Ibexa Cloud section](#ibexa-cloud-requirements-and-setup) for further reading on its requirements.

The minimal setup requires PHP,  MySQL/MariaDB, Apache/Nginx, Node.js and `yarn`.
Recommendation for production setups is to use Varnish/Fastly, Redis/Memcached, NFS/EFS/S3 and Solr/Elasticsearch in a [clustered setup](../guide/clustering.md).

> Using the latest supported version or each product or component is always recommended.

Reviewer: ```suggestion
Using the latest supported version of each product or component is always recommended.
```

Reviewer: Do we need to clarify "latest supported **by us**"? I mean, php 8 is supported by php but not by us.

Author: Probably, let's make it completely clear

Reviewer: This is covered by the "Using the latest listed version of each product or component is always recommended." I guess?

> Using the latest supported version or each product or component is always recommended.

Author: ```suggestion
Using the latest listed version of each product or component is always recommended.
```

## Operating system

- Debian 10.x "Buster"
- Ubuntu 20.04 "Focal Fossa"
- RHEL / CentOS 8.1+

## Web server

- Nginx 1.12, 1.14, 1.16
- Apache 2.4 (with required modules `mod_rewrite`, `mod_env` and recommended: `mod_setenvif`, `mod_expires`;
event MPM is recommended, if you need to use prefork you also need the `mod_php` module)

## DBMS

- MariaDB 10.3, 10.4 (optionally 10.2 - deprecated)
- MySQL 8.0 (optionally 5.7 - deprecated)
- PostgreSQL 10+

## PHP

- 7.3

> - 7.4

Reviewer: Should we add a "(recommended)" here for 7.4?

Reviewer: I realise now that the "Using the latest listed version of each product or component is always recommended." makes this kind of redundant. But it doesn't hurt either, now that it's there. 🤷‍♂️

- `php-xml`
- `php-mbstring`
- `php-json`
- `php-process` (on RHEL/CentOS)
- `php-intl`
- `php-curl`
- `php-pear` (optional, provides pecl)
- `php-gd` or `php-imagick` (via pecl on RHEL/CentOS)

### Cluster PHP packages

- `php-redis` or `php-memcached`

## Search

- Solr 7.7LTS or Solr 8
- Elasticsearch 7.7, using Oracle Java/Open JDK 8 or higher

## Graphic Handler

- GraphicsMagick
- ImageMagick
- GD

## [Clustering](../guide/clustering.md)

- Linux NFS or S3/EFS (for IO, aka binary files stored in content repository, not supported with legacy)
- Redis 4.0+, 5.0 or higher (separate instances for session and cache, both using a `volatile-*` [eviction policy](https://redis.io/topics/lru-cache), session instance configured for persistence) or [Memcached](https://memcached.org/) 1.5 or higher
- [Varnish](http://varnish-cache.org/) 6.0LTS with [varnish-modules](https://github.com/varnish/varnish-modules/blob/master/README.rst) or [Fastly](https://www.fastly.com/) using [the bundle provided with [[= product_name_exp =]]](../guide/http_cache.md#serving-varnish-through-fastly) (for HttpCache)

## Filesystem

- Linux ext4 / XFS

## Package manager

> - Composer 2.0.8 or higher

Reviewer: ```suggestion
- Composer 2.0.13 or higher
```
There's a security issue fixed in this version.

## Asset manager

- `Node.js` 10.22 or higher
- `yarn` 1.15.2 or higher

## Browser

> [[= product_name =]] is developed to work with *any* web browser that support modern standards, on *any* screen resolution suitable for web, running on *any* device. However for the Editorial and Administration User Interfaces you'll need; a minimum of 1366-by-768 screen resolution, a desktop or tablet device, and a recommended/supported browsers found below.

Reviewer: ```suggestion
[[= product_name =]] is developed to work with *any* web browser that support modern standards, on *any* screen resolution suitable for web, running on *any* device. However for the Editorial and Administration User Interfaces you'll need; a minimum of 1366-by-768 screen resolution, a desktop or tablet device, and a recommended/supported browser among the ones found below.
```

event MPM is recommended, if you need to use prefork you also need the `mod_php` module)

## DBMS

- MariaDB 10.3, 10.4 (optionally 10.2 - deprecated)
- MySQL 8.0 (optionally 5.7 - deprecated)
- PostgreSQL 10+

## PHP

- 7.3
- 7.4 (recommended)

### PHP packages

- `php-cli`
- `php-fpm`
- `php-mysql` (`php-mysqlnd`) or `php-pgsql`
- `php-xml`
- `php-mbstring`
- `php-json`
- `php-process` (on RHEL/CentOS)
- `php-intl`
- `php-curl`
- `php-pear` (optional, provides pecl)
- `php-gd` or `php-imagick` (via pecl on RHEL/CentOS)

### Cluster PHP packages

- `php-redis` or `php-memcached`

## Search

- Solr 7.7LTS or Solr 8
- Elasticsearch 7.7, using Oracle Java/Open JDK 8 or higher

## Graphic Handler

- GraphicsMagick
- ImageMagick
- GD

## [Clustering](../guide/clustering.md)

- Linux NFS or S3/EFS (for IO, aka binary files stored in content repository, not supported with legacy)
- Redis 4.0+, 5.0 or higher (separate instances for session and cache, both using a `volatile-*` [eviction policy](https://redis.io/topics/lru-cache), session instance configured for persistence) or [Memcached](https://memcached.org/) 1.5 or higher

> - [Varnish](http://varnish-cache.org/) 6.0LTS with [varnish-modules](https://github.com/varnish/varnish-modules/blob/master/README.rst) or [Fastly](https://www.fastly.com/) using [the bundle provided with [[= product_name_exp =]]](../guide/http_cache.md#serving-varnish-through-fastly) (for HttpCache)

Reviewer 2: ```suggestion
- [Varnish](http://varnish-cache.org/) 6.0LTS with [varnish-modules](https://github.com/varnish/varnish-modules/blob/master/README.md) or [Fastly](https://www.fastly.com/) using [the bundle provided with [[= product_name_exp =]]](../guide/http_cache.md#serving-varnish-through-fastly) (for HttpCache)
```

## Filesystem

- Linux ext4 / XFS

## Package manager

- Composer 2.0.13 or higher

## Asset manager

- `Node.js` 10.22 or higher
- `yarn` 1.15.2 or higher

## Browser

> [[= product_name =]] is developed to work with *any* web browser that support modern standards, on *any* screen resolution suitable for web, running on *any* device. However for the Editorial and Administration User Interfaces you'll need; a minimum of 1366-by-768 screen resolution, a desktop or tablet device, and a recommended/supported browser among the ones found below.

Reviewer 2: ```suggestion
[[= product_name =]] is developed to work with *any* web browsers that support modern standards, on *any* screen resolution suitable for web, running on *any* device. However for the Editorial and Administration User Interfaces you'll need; a minimum of 1366-by-768 screen resolution, a desktop or tablet device, and a recommended/supported browser among the ones found below.
```
